hydromt/components/grid.py

```diff
@@ -482,6 +443,7 @@ def data(self) -> xr.Dataset:
         """Model static gridded data as xarray.Dataset."""
         if self._data is None:
             self._initialize_grid()
+        assert self._data is not None
         return self._data
 
```

Comment:
The `set_crs` function could be a `@crs.setter`. Should we only have this for `GridComponent`, or should every `SpatialModelComponent` be able to do this?